Add missing knowledge checks and learning outcomes to ruby/object-oriented-programming lesson (#23039)
TheOdinProject/curriculum: ruby_programming/object_oriented_programming_basics/lesson_oop.md

* Now you're ready to read through [Zetcode's OOP section](http://zetcode.com/lang/rubytutorial/oop/).
* Read through [Zetcode's second OOP section](http://zetcode.com/lang/rubytutorial/oop2/) until they start talking about exceptions (~80% of the way down).

### Knowledge Check

This section contains questions for you to check your understanding of this lesson. If you’re having trouble answering the questions below on your own, review the material above to find the answer.

- <a class="knowledge-check-link" href="https://www.eriktrautman.com/posts/ruby-explained-classes">What is a class?</a>

> - <a class="knowledge-check-link" href="https://www.eriktrautman.com/posts/ruby-explained-classes">When should you use a class?</a>

Reviewer: Are these questions directly answered in the linked resource? I'm not seeing them but I 100% could just be missing them.

Author: Not directly answered. Eriks post indirectly answers it. I'll update the links to a more directly answered resource.

* Now you're ready to read through [Zetcode's OOP section](http://zetcode.com/lang/rubytutorial/oop/).
* Read through [Zetcode's second OOP section](http://zetcode.com/lang/rubytutorial/oop2/) until they start talking about exceptions (~80% of the way down).

### Knowledge Check

This section contains questions for you to check your understanding of this lesson. If you’re having trouble answering the questions below on your own, review the material above to find the answer.

- <a class="knowledge-check-link" href="https://launchschool.com/books/oo_ruby/read/the_object_model#classesdefineobjects">What is a class?</a>
- <a class="knowledge-check-link" href="https://www.eriktrautman.com/posts/ruby-explained-classes">What are getters and setters?</a>
- <a class="knowledge-check-link" href="https://www.eriktrautman.com/posts/ruby-explained-classes">What is inheritance?</a>
- <a class="knowledge-check-link" href="https://www.eriktrautman.com/posts/ruby-explained-inheritance-and-scope">What is a scope?</a>
- <a class="knowledge-check-link" href="https://www.eriktrautman.com/posts/ruby-explained-inheritance-and-scope">When is a new scope defined?</a>

> - <a class="knowledge-check-link" href="https://launchschool.com/books/oo_ruby/read/classes_and_objects_part1#instancevariables">What do we use instance variables for?</a>
> - <a class="knowledge-check-link" href="https://www.eriktrautman.com/posts/ruby-explained-classes">When should you not use class variables?</a>

Reviewer 2: Thanks for looking for where these two questions would be answered in this LONG lesson! After reviewing these resources, I do not think my proposed "why they should not use global or class variables" is really explained (no wonder people get confused here). So I've adjusting the wording to what I think that resource actually does explain. 

In addition, as a nit pick, I suggested a change for the instance variable one, simply because I do like like the shift of wording here to using "we" (in general I don't like the "we" wording, like "How are we doing today?" but I realize that I am could be in the minority on this, so this is just simply a suggestion.

```suggestion
- <a class="knowledge-check-link" href="https://launchschool.com/books/oo_ruby/read/classes_and_objects_part1#instancevariables">What is the purpose of instance variables?</a>
- <a class="knowledge-check-link" href="https://www.eriktrautman.com/posts/ruby-explained-classes">What is the difference between class variables and instance variables?</a>
```
